core/src/main/java/org/elasticsearch/cluster/routing/allocation/RerouteExplanation.java

```diff
@@ -52,23 +52,21 @@ public Decision decisions() {
     }
 
     public static RerouteExplanation readFrom(StreamInput in) throws IOException {
-        String commandName = in.readString();
-        AllocationCommand command = AllocationCommands.lookupFactorySafe(commandName).readFrom(in);
+        AllocationCommand command = in.readAllocationCommand();
```

Comment:
I don't follow. As the `NamedWriteable` interface is public, it is trivial for anyone to write the follow method and put it in a Utils class:

```
public static void writeNamedWriteable(StreamOutput out, NamedWriteable<?> namedWriteable) throws IOException {
    out.writeString(namedWriteable.getWriteableName());
    namedWriteable.writeTo(out);
}
```

I feel that adding all these methods to StreamInput / StreamOutput is the *wrong* tradeoff to do. We should strive to modularize our code better. We can discuss this on Fixit Friday. I don't want to block this PR on that topic.